clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -522,7 +522,12 @@ private long waitOnMetadata(String topic, long maxWaitMs) throws InterruptedExce
             log.trace("Requesting metadata update for topic {}.", topic);
             int version = metadata.requestUpdate();
             sender.wakeup();
-            metadata.awaitUpdate(version, remainingWaitMs);
+            try {
+                metadata.awaitUpdate(version, remainingWaitMs);
```

Comment:
It's a public method, true, but `Metadata` is not considered public API (it's annoying that we don't make this crystal clear via package names). Still, you're probably right that it's not worth potentially breaking people for this small improvement.